src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/judgeapproval/ApproveDraftOrdersAboutToSubmitHandler.java

```diff
@@ -0,0 +1,59 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.mapper.FinremCaseDetailsMapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.DraftOrdersWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.judgeapproval.ApproveOrderService;
+
+@Slf4j
+@Service
+public class ApproveDraftOrdersAboutToSubmitHandler extends FinremCallbackHandler {
+
+    private final ApproveOrderService approveOrderService;
+
+    public ApproveDraftOrdersAboutToSubmitHandler(FinremCaseDetailsMapper finremCaseDetailsMapper, ApproveOrderService approveOrderService) {
+        super(finremCaseDetailsMapper);
+        this.approveOrderService = approveOrderService;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.APPROVE_ORDERS.equals(eventType);
+    }
+
+    @Override
+    public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
+                                                                              String userAuthorisation) {
+        FinremCaseDetails caseDetails = callbackRequest.getCaseDetails();
+        String caseId = String.valueOf(caseDetails.getId());
+        log.info("Invoking contested {} about-to-submit event callback for Case ID: {}", callbackRequest.getEventType(), caseId);
+
+        FinremCaseData finremCaseData = caseDetails.getData();
+        DraftOrdersWrapper draftOrdersWrapper = finremCaseData.getDraftOrdersWrapper();
+        approveOrderService.populateJudgeDecisions(draftOrdersWrapper, userAuthorisation);
+        clearInputFields(draftOrdersWrapper);
+
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder().data(finremCaseData).build();
+    }
+
+    private void clearInputFields(DraftOrdersWrapper draftOrdersWrapper) {
+        draftOrdersWrapper.setJudgeApproval1(null);
+        draftOrdersWrapper.setJudgeApproval2(null);
+        draftOrdersWrapper.setJudgeApproval3(null);
+        draftOrdersWrapper.setJudgeApproval4(null);
+        draftOrdersWrapper.setJudgeApproval5(null);
+        draftOrdersWrapper.setHearingInstruction(null);
+        draftOrdersWrapper.setShowWarningMessageToJudge(null);
+    }
+}
```

Comment:
In a situation where the callback response fails, it looks like clearInputFields may have already removed the data.  If that happened and the judge clicked submit again, could they submit with empty answers?

If you think this could be an issue, consider moving  clearInputFields to a submitted callback. Or add a log, then create a ticket for some follow up work to do this later.

Reply:
submitted callback is not an option since we cannot change the data anymore.